### PR TITLE
Add support for multi-column rows to EnforceSingleRowOperator

### PR DESCRIPTION
**Pull request by martint**
The operator is currently hard-coded to a single column. This extends
it to support any number of columns.

### CODE REVIEW

File: core/trino-main/src/main/java/io/trino/operator/EnforceSingleRowOperator.java [hunk 1]
```diff
@@ -32,20 +33,39 @@
     {
         private final int operatorId;
         private final PlanNodeId planNodeId;
+        private final Page nullValuePage;
```
Comment (kokosing) [inline core/trino-main/src/main/java/io/trino/operator/EnforceSingleRowOperator.java:36]: nullValuesPage

**Review by kokosing — COMMENTED** [review]
Why do you need to pass types here?

**Comment by martint** [comment]
> Why do you need to pass types here?

Because the types are the ones who know how to instantiate a block. Technically, the previous implementation was incorrect, as it was creating a block that might be incompatible with certain types.

**Comment by martint** [comment]
@kokosing, updated

File: core/trino-main/src/main/java/io/trino/operator/EnforceSingleRowOperator.java [hunk 2]
```diff
@@ -32,20 +33,39 @@
...
+        }
+
+        private EnforceSingleRowOperatorFactory(int operatorId, PlanNodeId planNodeId, Page nullValuesPage)
         {
             this.operatorId = operatorId;
             this.planNodeId = requireNonNull(planNodeId, "planNodeId is null");
+            this.nullValuesPage = requireNonNull(nullValuesPage, "nullValuesPage is null");
+        }
+
+        private static Page makeNullValuePage(List<Type> types)
```
Comment (kokosing) [inline core/trino-main/src/main/java/io/trino/operator/EnforceSingleRowOperator.java:52]: makeNullValuesPage

**Review by kokosing — APPROVED** [review]

**Pull request merged**